pulp/apis/highs_api.py

```diff
@@ -0,0 +1,163 @@
+# PuLP : Python LP Modeler
+# Version 2.4
+
+# Copyright (c) 2002-2005, Jean-Sebastien Roy (js@jeannot.org)
+# Modifications Copyright (c) 2007- Stuart Anthony Mitchell (s.mitchell@auckland.ac.nz)
+# $Id:solvers.py 1791 2008-04-23 22:54:34Z smit023 $
+
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE."""
+
+# Modified by Sam Mathew (@samiit on Github)
+# Users would need to install HiGHS on their machine and provide the path to the executable. Please look at this thread: https://github.com/ERGO-Code/HiGHS/issues/527#issuecomment-894852288
+# More instructions on: https://www.highs.dev
+
+from .core import LpSolver_CMD, subprocess, PulpSolverError
+import os, sys
+from .. import constants
+import warnings
+
+
+class HiGHS_CMD(LpSolver_CMD):
+    """The HiGHS_CMD solver"""
+    name = 'HiGHS_CMD'
+
+    def __init__(self, path=None, keepFiles=False, mip=True, msg=True, options=None,  timeLimit=None):
+        """
+        :param bool mip: if False, assume LP even if integer variables
+        :param bool msg: if False, no log is shown
+        :param float timeLimit: maximum time for solver (in seconds)
+        :param list options: list of additional options to pass to solver
+        :param bool keepFiles: if True, files are saved in the current directory and not deleted after solving
+        :param str path: path to the solver binary (you can get binaries for your platform from https://github.com/JuliaBinaryWrappers/HiGHS_jll.jl/releases, or else compile from source - https://highs.dev)
+        """
+        LpSolver_CMD.__init__(self, mip=mip, msg=msg, timeLimit=timeLimit,
+                              options=options, path=path, keepFiles=keepFiles)
+
+    def defaultPath(self):
+        return self.executableExtension("")
```

Comment:
here you can put the name of the executable in case it's in the `PATH`. This way the user does not need to provide the path to the solver as argument. For example, you could put:

```python
def defaultPath(self):
    self.executableExtension("highs")
```